web/src/components/upgrade_service/AppConfig.tsx

```diff
@@ -59,31 +50,23 @@ type ConfigGroupItem = {
 
 type RequiredItems = string[];
 
-type State = {
-  activeGroups: string[];
-  configErrorMessage: string;
-  configGroups: ConfigGroup[];
-  configLoading: boolean;
-  displayErrorModal: boolean;
-  errorTitle: string;
-  gettingConfigErrMsg: string;
-  initialConfigGroups: ConfigGroup[];
-  showConfigError: boolean;
-  showValidationError: boolean;
-};
-
 const validationErrorMessage =
   "Error detected. Please use config nav to the left to locate and fix issues.";
 
-class AppConfig extends Component<Props, State> {
-  sidebarWrapper: HTMLElement;
-
-  fetchController: AbortController | null;
+export const AppConfig = ({
```

Comment:
converted into functional component so we can utilize hooks. will need to test this component out more in case i missed anything.